packages/react/src/components/DateTimePicker/DateTimePickerV2.jsx

```diff
@@ -719,17 +662,26 @@ const DateTimePicker = ({
         className={`${iotPrefix}--date-time-pickerv2__wrapper`}
         style={{ '--wrapper-width': hasIconOnly ? '3rem' : '20rem' }}
         role="button"
-        tabIndex="0"
-        onKeyDown={(event) => {
-          if (event.key === 'Enter') {
-            setIsFlyoutOpen((flyoutOpen) => !flyoutOpen);
-            // have to do this to stop the mouse click from being fired
-            event.preventDefault();
+        onClick={onFieldInteraction}
+        onKeyDown={handleSpecificKeyDown(['Enter', ' ', 'Escape', 'ArrowDown'], (event) => {
```

Comment:
Ah, got it. No, I don't think it's a concern. The handleSpecificKeyDown is going to return a new function every time it's called. So, even if we memoized it you're going to end up with a new function being created every time anyway and the memoizing just adds another allocation. I could be wrong, but I think the re-usability (and readability) of the curried function wins out over the minor optimization improvements.

Reply:
I was just wondering about having it behind one more function. Since we are making a function call with `handleSpecificKeydown` and that will be ran on every render not just on every keydown.

Reply:
But to your point its creating a new function anyways so may be the same as the anonymous function. 🤔

Reply:
I'm not I follow. You're saying you'd prefer it like this: 

```jsx
onKeyDown={(event) => {
  handleSpecificKeyDown(['Enter', ' ', 'Escape', 'ArrowDown'], (event) => {
      ...
  })(event);
}
```

Reply:
@davidicus Friendly bump here ☝️

Reply:
I was questioning whether that would prevent the creation of the new function that `handleSpecificKeyDown` returns on each render but it would do the same.